lib/core/wallet_loading_service.dart

```diff
@@ -46,6 +53,20 @@ class WalletLoadingService {
       await updateMoneroWalletPassword(wallet);
     }
 
+    TorConnectionMode mode = TorConnectionMode.deserialize(
+      raw: sharedPreferences.getInt(PreferencesKey.currentTorConnectionModeKey) ?? 0,
+    );
+
+    if ([WalletType.bitcoin, WalletType.litecoin].contains(type)) {
+      bitcoin!.setTorOnly(wallet, mode == TorConnectionMode.torOnly);
+    }
+
+    final status = torViewModel.torConnectionStatus;
+    if (status == TorConnectionStatus.connected || status == TorConnectionStatus.connecting) {
+      // connect the node to the tor proxy:
+      await torViewModel.startTor();
+    }
```

Comment:
start tor if already connected or connecting?

Reply:
the comment explains what's happening here (we call this just to connect the node to the proxy)
but I agree it should be more clear and I just forgot that I extracted that out to it's own function so the latest commit just calls that function directly